smithy-aws-protocol-tests/model/rpcV2/cbor-structs.smithy

```diff
@@ -319,4 +404,167 @@ structure SimpleScalarStructure {
+            floatValue: 7.624,
+            integerValue: 256,
+            longValue: 9873,
+            shortValue: 9898,
+            stringValue: "simple",
+            trueBooleanValue: true
+        },
+        appliesTo: "server"
+    },
+    {
+        id: "RpcV2CborSomeDefaultValuesLeftOut",
+        documentation: "Left over default values should be filled in.",
+        protocol: rpcv2Cbor,
+        method: "POST",
+        uri: "/service/RpcV2Protocol/operation/DefaultFieldInputOutput",
+        // http://ec2-54-84-9-83.compute-1.amazonaws.com/hex?value=qWlieXRlVmFsdWUAa2RvdWJsZVZhbHVl%2B0AHHKwIMSbpcWZhbHNlQm9vbGVhblZhbHVl9WpmbG9hdFZhbHVl%2BkBPXClsaW50ZWdlclZhbHVlGDhpbG9uZ1ZhbHVlGSaRanNob3J0VmFsdWUZJqprc3RyaW5nVmFsdWVmc2ltcGxlcHRydWVCb29sZWFuVmFsdWX0
+        body: "qWlieXRlVmFsdWUAa2RvdWJsZVZhbHVl+0AHHKwIMSbpcWZhbHNlQm9vbGVhblZhbHVl9WpmbG9hdFZhbHVl+kBPXClsaW50ZWdlclZhbHVlGDhpbG9uZ1ZhbHVlGSaRanNob3J0VmFsdWUZJqprc3RyaW5nVmFsdWVmc2ltcGxlcHRydWVCb29sZWFuVmFsdWX0"
+        bodyMediaType: "application/cbor",
+        headers: {
+            "smithy-protocol": "rpc-v2-cbor",
+            "Accept": "application/cbor",
+            "Content-Type": "application/cbor"
+        },
+        params: {
+            byteValue: 0,
+            doubleValue: 2.889,
+            falseBooleanValue: true,
+            floatValue: 3.24,
+            integerValue: 56,
+            trueBooleanValue: false
+        },
+    },
+    {
+        id: "RpcV2CborServerShouldFillLeftoutDefaultValues",
+        documentation: "The server should populate the remaining default values when it receives
+            a request that lacks values for fields with defaults defined in the model.",
+        protocol: rpcv2Cbor,
+        method: "POST",
+        uri: "/service/RpcV2Protocol/operation/DefaultFieldInputOutput",
+        // http://ec2-54-84-9-83.compute-1.amazonaws.com/hex?value=pmlieXRlVmFsdWUAa2RvdWJsZVZhbHVl%2B0AHHKwIMSbpcWZhbHNlQm9vbGVhblZhbHVl9WpmbG9hdFZhbHVl%2BkBPXClsaW50ZWdlclZhbHVlGDhwdHJ1ZUJvb2xlYW5WYWx1ZfQ%3D
+        body: "pmlieXRlVmFsdWUAa2RvdWJsZVZhbHVl+0AHHKwIMSbpcWZhbHNlQm9vbGVhblZhbHVl9WpmbG9hdFZhbHVl+kBPXClsaW50ZWdlclZhbHVlGDhwdHJ1ZUJvb2xlYW5WYWx1ZfQ="
+        bodyMediaType: "application/cbor",
+        headers: {
+            "smithy-protocol": "rpc-v2-cbor",
+            "Accept": "application/cbor",
+            "Content-Type": "application/cbor"
+        },
+        params: {
+            byteValue: 0,
+            doubleValue: 2.889,
+            falseBooleanValue: true,
+            floatValue: 3.24,
+            integerValue: 56,
+            longValue: 9873,
+            shortValue: 9898,
+            stringValue: "simple",
+            trueBooleanValue: false
+        },
+        appliesTo: "server"
+    }
+])
+
+apply DefaultFieldInputOutput @httpResponseTests([
+    {
+        id: "RpcV2CborEmptyOutputShouldSerializeDefaultValues",
+        documentation: "Serializes an empty output with default values filled in.",
```

Comment:
I assume this is meant to say deserializes instead (since it's an output test without `appliesTo` server).

Reply:
This test is applicable to both the server and the client. The [ServerProtocolTestGenerator](https://github.com/smithy-lang/smithy-rs/blob/main/codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt#L160) picks up tests that are marked either with `server` only or that have an empty `appliesTo` field. The documentation has been updated to clarify this.